data_source/gateway.go

```diff
@@ -27,14 +27,144 @@ func (g *Gateway) GetBlockByNumber(blockNumber uint64) (*core.Block, error) {
 
 // GetTransaction gets the transaction for a given transaction hash from the feeder gateway,
 // then adapts it to the appropriate core.Transaction types.
-func (g *Gateway) GetTransaction(transactionHash *felt.Felt) (*core.Transaction, error) {
-	return nil, errors.New("not implemented")
+func (g *Gateway) GetTransaction(transactionHash *felt.Felt) (core.Transaction, error) {
+	response, err := g.client.GetTransaction(transactionHash)
+	if err != nil {
+		return nil, err
+	}
+
+	tx, err := adaptTransaction(response, g)
+	if err != nil {
+		return nil, err
+	}
+
+	return tx, nil
+}
+
+func adaptTransaction(response *clients.TransactionStatus, g *Gateway) (core.Transaction, error) {
+	txType := response.Transaction.Type
+	switch txType {
+	case "DECLARE":
+		class, err := g.GetClass(response.Transaction.ClassHash)
+		if err != nil {
+			return nil, err
+		}
+		declareTx, err := adaptDeclareTransaction(response, class)
+		if err != nil {
+			return nil, err
+		}
+		return declareTx, nil
+	case "DEPLOY":
+		class, err := g.GetClass(response.Transaction.ClassHash)
```

Comment:
It'll help avoid a lot of repetition as well.

```suggestion
	txType := transaction.Type
	switch txType {
	case "DECLARE":
		class, err := g.GetClass(transaction.ClassHash)
		if err != nil {
			return nil, err
		}
		declareTx, err := adaptDeclareTransaction(transaction, class)
		if err != nil {
			return nil, err
		}
		return declareTx, nil
	case "DEPLOY":
		class, err := g.GetClass(transaction.ClassHash)
```